### PR TITLE
Update cluster-resources.md

### PR DESCRIPTION
**Pull request by LavredisG**
/kind documentation

Some parts were translated from the original Chinese text to be more accurate and
easier to understand for the reader.

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
/assign

**Comment by LavredisG** [comment]
> /assign

Also, on the page's last example, the Pod requires 3C, 20Gi and it is stated that every node from Grade 2 and above meets this requirement. However, if a node has 2C and 16Gi available, it will be classified as Grade 2 without being able to provide the resources needed to the pod. Is that right or I am misunderstanding something?

File: docs/userguide/scheduling/cluster-resources.md [hunk 1]
```diff
@@ -104,10 +103,11 @@ Considering the amount of available resources, the scheduler prefers to schedule
...
-From the `ResourceSummary`, we get that there are 2000 core CPU left for the cluster, but actually, this cluster cannot run any pod that requires CPU greater than 1 core.
+However, the `ResourceSummary` is not precise enough, it mechanically counts the resources on all nodes, but ignores the fragmented resources of these nodes. For example, a cluster with 2000 nodes has only 1 core CPU left on each node.
+From the `ResourceSummary`, we get that there are 2000 cores CPU left for the cluster, but actually, this cluster cannot run any pod that requires more than 1 core CPU.
 
-Therefore, we introduce a `CustomizedClusterResourceModeling` for each cluster that records the resource portrait of each node. Karmada will collect node and pod information for each cluster. After calculation, this node will be divided into the appropriate resource model configured by the users.
+Therefore, we introduce a `CustomizedClusterResourceModeling` for each cluster that records the resource profile of each node.
+Karmada collects node and pod information from each cluster and computes the appropriate -user configured- resource model to categorize the node into.
```
Comment (RainbowMango) [inline docs/userguide/scheduling/cluster-resources.md:110]:
```suggestion
Karmada collects node and pod information from each cluster and computes the appropriate `user configured` resource model to categorize the node info.
```
Is it a typo here?

**Comment by RainbowMango** [comment]
> However, if a node has 2C and 16Gi available, it will be classified as Grade 2 without being able to provide the resources needed to the pod. Is that right or I am misunderstanding something?

Yeah, you are right! Nice finding!
It seems grade 2 doesn't fulfill the request and shouldn't be counted. And I checked the code that, only when the minimum value > `requested value`, the node will be counted. See [the code here](https://github.com/karmada-io/karmada/blob/e9e8653c8898605ec57c7e006293864657ef1bc5/pkg/estimator/client/general.go#L222).

I guess the documentation might be incorrect. @chaosi-zju can you help to confirm that?

**Comment by RainbowMango** [comment]
> In addition to these, docs state that "when a Pod needs to be scheduled to a specific cluster, they (scheduler) will compare the number of nodes in the model that satisfy the resource request of the Pod in different clusters, and schedule the Pod to the cluster with more nodes". But when we look at the numerical example, the choice is being made by choosing the cluster that could accommodate as many instances of that Pod (hence member3 is chosen in the first example), because if it chose based on the number of nodes able to accomodate the pod, it would have to choose member2 which has 8 nodes instead of member3 with 1 node (assuming that all of Grade 2 nodes can meet pod's requirements in member 2 as noted above).

Yes, you are right. This is not just about checking the number of nodes, but calculating which cluster can actually accommodate the most Pods.

**Comment by RainbowMango** [comment]
@LavredisG I think we can fix the incorrect docs after @chaosi-zju confirms it by another PR, let this PR focus on gramma issues.

File: docs/userguide/scheduling/cluster-resources.md (hunk 1 above)
Comment (LavredisG) [reply docs/userguide/scheduling/cluster-resources.md:110]: It was meant to say that a node is being categorized into a resource model.

**Comment by LavredisG** [comment]
> > In addition to these, docs state that "when a Pod needs to be scheduled to a specific cluster, they (scheduler) will compare the number of nodes in the model that satisfy the resource request of the Pod in different clusters, and schedule the Pod to the cluster with more nodes". But when we look at the numerical example, the choice is being made by choosing the cluster that could accommodate as many instances of that Pod (hence member3 is chosen in the first example), because if it chose based on the number of nodes able to accomodate the pod, it would have to choose member2 which has 8 nodes instead of member3 with 1 node (assuming that all of Grade 2 nodes can meet pod's requirements in member 2 as noted above).
> 
> Yes, you are right. This is not just about checking the number of nodes, but calculating which cluster can actually accommodate the most Pods.

Thank you for the clarifications!

**Comment by LavredisG** [comment]
> > However, if a node has 2C and 16Gi available, it will be classified as Grade 2 without being able to provide the resources needed to the pod. Is that right or I am misunderstanding something?
> 
> Yeah, you are right! Nice finding! It seems grade 2 doesn't fulfill the request and shouldn't be counted. And I checked the code that, only when the minimum value > `requested value`, the node will be counted. See [the code here](https://github.com/karmada-io/karmada/blob/e9e8653c8898605ec57c7e006293864657ef1bc5/pkg/estimator/client/general.go#L222).
> 
> I guess the documentation might be incorrect. @chaosi-zju can you help to confirm that?

Oh ok, so even if some nodes from Grade 2 can accommodate the pod (those with more than 3C and 20Gi), we don't count them in this case because they have to meet the demands with their Min values, is that correct?

File: docs/userguide/scheduling/cluster-resources.md (hunk 1 above)
Comment (RainbowMango) [reply docs/userguide/scheduling/cluster-resources.md]: Oh. I get it. Thanks.

**Comment by RainbowMango** [comment]
> Oh ok, so even if some nodes from Grade 2 can accommodate the pod (those with more than 3C and 20Gi), we don't count them in this case because they have to meet the demands with their Min values, is that correct?

Yes, nodes from Grade 2 would be ignored as we are not sure all nodes in this grade fulfill the request.

**Comment by LavredisG** [comment]
> > I think we can fix the incorrect docs after @chaosi-zju confirms it by another PR
> 
> So, is there any line in previous doc should be modified?

I'd say both the numerical examples and the part about choosing the cluster with the most nodes satisfying the request should be modified ("the cluster that can accommodate the most pod replicas" should be the right case)

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

As discussed above, we can have the following PR to correct the mistakes.
@LavredisG Let me know if you'd like to help with that.

**Comment by RainbowMango** [comment]
Opened #715 to track this. The cn doc will be updated after en doc.

**Comment by RainbowMango** [comment]
Just a short update:
After further investigation, I think it is a bug and sent a PR https://github.com/karmada-io/karmada/pull/5706.

**Pull request merged**